src/sentry/api/endpoints/group_notes.py

```diff
@@ -74,11 +75,20 @@ def post(self, request: Request, group) -> Response:
             group=group, user_ids=mentioned_users["users"], reason=GroupSubscriptionReason.mentioned
         )
 
-        GroupSubscription.objects.bulk_subscribe(
-            group=group,
-            user_ids=mentioned_users["team_users"],
-            reason=GroupSubscriptionReason.team_mentioned,
-        )
+        if features.has(
+            "organizations:team-workflow-notifications", group.organization, request.user
+        ):
+            GroupSubscription.objects.bulk_subscribe(
+                group=group,
+                teams=mentioned_users["teams"],
```

Comment:
if we're using team_ids for bulk_subscribe,  this and the return of the mention function need to also be changed.